spyder/preferences/languageserver.py

```diff
@@ -608,30 +607,356 @@ def mouseDoubleClickEvent(self, event):
+        code_style_g_layout.addWidget(
+            self.code_style_filenames_match.label, 1, 0)
+        code_style_g_layout.addWidget(
+            self.code_style_filenames_match.textbox, 1, 1)
+        code_style_g_layout.addWidget(self.code_style_exclude.label, 2, 0)
+        code_style_g_layout.addWidget(self.code_style_exclude.textbox, 2, 1)
+        code_style_g_layout.addWidget(code_style_select.label, 3, 0)
+        code_style_g_layout.addWidget(code_style_select.textbox, 3, 1)
+        code_style_g_layout.addWidget(code_style_ignore.label, 4, 0)
+        code_style_g_layout.addWidget(code_style_ignore.textbox, 4, 1)
+        code_style_g_layout.addWidget(code_style_max_line_length.plabel, 5, 0)
+        code_style_g_layout.addWidget(
+            code_style_max_line_length.spinbox, 5, 1)
+
+        # Set Code style options enabled/disabled
+        code_style_g_widget = QWidget()
+        code_style_g_widget.setLayout(code_style_g_layout)
+        code_style_g_widget.setEnabled(self.get_option('pycodestyle'))
+        code_style_check.toggled.connect(code_style_g_widget.setEnabled)
+
+        # Code style layout
+        code_style_layout = QVBoxLayout()
+        code_style_layout.addWidget(code_style_label)
+        code_style_layout.addWidget(code_style_check)
+        code_style_layout.addWidget(code_style_g_widget)
+
+        code_style_widget = QWidget()
+        code_style_widget.setLayout(code_style_layout)
+
+        # --- Docstring tab ---
+        # Docstring style label
+        numpy_url = (
+            "<a href='https://numpydoc.readthedocs.io/en/"
+            "latest/format.html'>Numpy</a>")
+        pep257_url = (
+            "<a href='https://www.python.org/dev/peps/pep-0257/'>Pep 257</a>")
+        docstring_style_codes = _(
+            "<a href='http://www.pydocstyle.org/en/stable"
+            "/error_codes.html'>page</a>")
+        docstring_style_label = QLabel(
+            _("Here you can decide if you want to perform style analysis on "
+              "your docstrings according to the {} or {} conventions. You can "
+              "also decide if you want to show or ignore specific errors, "
+              "according to the codes found on this {}.").format(
+                  numpy_url, pep257_url, docstring_style_codes))
+        docstring_style_label.setOpenExternalLinks(True)
+        docstring_style_label.setWordWrap(True)
+
+        # Docstring style checkbox
+        docstring_style_check = self.create_checkbox(
+            _("Enable docstring style linting"),
+            'pydocstyle')
+
+        # Docstring style options
+        docstring_style_convention = self.create_combobox(
+            _("Choose the convention used to lint docstrings: "),
+            (("Numpy", 'numpy'),
+             ("Pep 257", 'pep257')),
+            'pydocstyle/convention')
+        docstring_style_add_select = self.create_lineedit(
+            _("Display the following errors in addition to the "
```

Comment:
Consistency. ``Show the following errors:``

Reply:
That does make it clearer for PEP 257 and Numpydoc. My concern was that it doesn't make 100% sense for custom since it isn't in addition to anything.

Reply:
Yes, that's why this field is going to be disabled when `Custom` is active.